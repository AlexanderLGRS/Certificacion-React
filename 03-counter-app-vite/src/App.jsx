import { Buffer } from 'buffer';
import PropTypes from 'prop-types'
import crypto from "crypto";

function App({ title, subtitle, name }) {
  const getValue = (a, b) => {
    return a + b
  }

  return (
    <div className="App">
      <h1 onClick={() => {  }}>{title}</h1>
      <h1>{subtitle}</h1>
      <h1>{name}</h1>
      <h1>{getValue(122, 184)}</h1>
    </div>
  )
}

export default App

App.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string,
  name: PropTypes.string
}
App.defaultProps = {
  title: 'Titulo por defecto',
  subtitle: 'Subtitulo por defecto',
  name: 'Nombre por defecto',
}