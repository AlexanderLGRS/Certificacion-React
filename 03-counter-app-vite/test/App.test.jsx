import { render } from "@testing-library/react"
import App from "../src/App"

describe('Pruebas en App', () => {
    test('Debe hacer match con el snapshot', () => {
        const { container } = render(<App title='Mi titulo' />)
        console.log(container);
        expect(container).toMatchSnapshot()
    })
})