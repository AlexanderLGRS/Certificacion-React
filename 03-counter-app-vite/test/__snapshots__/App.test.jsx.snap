// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en App Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="App"
  >
    <h1>
      Mi titulo
    </h1>
    <h1>
      Subtitulo por defecto
    </h1>
    <h1>
      Nombre por defecto
    </h1>
    <h1>
      306
    </h1>
  </div>
</div>
`;
